aspnetcore/mvc/models/validation.md

```diff
@@ -72,6 +72,41 @@ When you use a [Nullable\<T> type](/dotnet/csharp/programming-guide/nullable-typ
 
 Client-side validation requires a value for a form field that corresponds to a model property that you've marked `Required` and for a non-nullable type property that you haven't marked `Required`. `Required` can be used to control the client-side validation error message.
 
+::: moniker range=">= aspnetcore-2.1"
+
+## Top-level node validation
+
+Top-level nodes include:
+
+* Action parameters
+* Controller properties
+* Page handler parameters
+* Page model properties
+
+Model-bound top-level nodes are validated in addition to validating model properties. In the following example from the sample app, the `VerifyPhone` method uses the <xref:System.ComponentModel.DataAnnotations.RegularExpressionAttribute> to validate the user data in the Phone field of a form:
+
+[!code-csharp[](validation/sample/UsersController.cs?name=snippet_VerifyPhone)]
+
+Top-level nodes can use <xref:Microsoft.AspNetCore.Mvc.ModelBinding.BindRequiredAttribute> with validation attributes. In the following example, `name` is required and bound from the route, and `count` is required and bound from the query string:
+
+```csharp
+public IActionResult PostData(
```

Comment:
Can you move these two snippets to a file? Also, can you add a form to post and test? I think the sample at https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/search?view=aspnetcore-2.2 has everything you need.